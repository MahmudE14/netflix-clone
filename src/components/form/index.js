import { Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form'

export default function Form({ children, ...restProps }) {
    return (
        <Container {...restProps}>{children}</Container>
    )
};

Form.Base = function FormContainer({ children, ...restProps }) {
    return <Base { ...restProps }>{children}</Base>
}

Form.Error = function FormContainer({ children, ...restProps }) {
    return <Error { ...restProps }>{children}</Error>
}

Form.Title = function FormContainer({ children, ...restProps }) {
    return <Title { ...restProps }>{children}</Title>
}

Form.Text = function FormContainer({ children, ...restProps }) {
    return <Text { ...restProps }>{children}</Text>
}

Form.TextSmall = function FormContainer({ children, ...restProps }) {
    return <TextSmall { ...restProps }>{children}</TextSmall>
}

Form.Link = function FormContainer({ children, ...restProps }) {
    return <Link { ...restProps }>{children}</Link>
}

Form.Input = function FormContainer({ children, ...restProps }) {
    return <Input { ...restProps }>{children}</Input>
}

Form.Submit = function FormContainer({ children, ...restProps }) {
    return <Submit { ...restProps }>{children}</Submit>
}
