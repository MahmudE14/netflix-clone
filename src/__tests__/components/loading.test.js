import { getByText, render, screen } from "@testing-library/react";
import { Loading } from "../../components";

describe("<Loading />", () => {
    it("render the <Loading /> component", () => {
      const { container } = render(
        <Loading src="/images/misc/loading.gif" data-testid="loading" />
      )

      expect(screen.getByTestId("loading")).toBeTruthy()
      expect(screen.getByTestId("loading-picture")).toBeTruthy()
      expect(container.firstChild).toMatchSnapshot()
    })

    it('renders the <Loading.ReleaseBody />', () => {
        const { container } = render(<Loading.ReleaseBody />)
        expect(container.firstChild).toMatchSnapshot()
    })
})
