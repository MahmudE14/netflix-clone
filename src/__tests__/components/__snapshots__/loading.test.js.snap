// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> render the <Loading /> component 1`] = `
<div
  class="loading__Spinner-sc-1j6l9vs-0 eGAuuN"
  data-testid="loading"
>
  <img
    class="loading__Picture-sc-1j6l9vs-1 jOEqss"
    data-testid="loading-picture"
    src="/images/users//images/misc/loading.gif.png"
  />
</div>
`;

exports[`<Loading /> renders the <Loading.ReleaseBody /> 1`] = `null`;
